pkg/statistics/fmsketch.go

```diff
@@ -88,27 +99,28 @@ func (s *FMSketch) NDV() int64 {
 	if s == nil {
 		return 0
 	}
-	// The size of the mask (incremented by one) is 2^r, where r is the maximum number of trailing zeroes observed in the hashed values.
-	// The count of unique hashed values is the number of unique elements in the hashset.
-	// This estimation method is based on the Flajolet-Martin algorithm for estimating the number of distinct elements in a stream.
+	// The estimated count of distinct values is 2^r * count, where 'r' is the maximum number of trailing zeroes observed and 'count' is the number of unique hashed values.
+	// The fundamental idea is that the hash function maps the input domain onto a logarithmic range.
+	// So the count of distinct values with 'r' trailing zeroes is n / 2^r, where 'n' is the number of distinct values.
+	// Therefore, the estimated count of distinct values is 2^r * count = n.
 	return int64(s.mask+1) * int64(s.hashset.Count())
 }
 
 // insertHashValue inserts a hashed value into the sketch.
 func (s *FMSketch) insertHashValue(hashVal uint64) {
 	// If the hashed value is already in the sketch (determined by bitwise AND with the mask), return without inserting.
-	// This is because the number of trailing zeroes in the hashed value is less than or equal to the mask value.
+	// This is because the number of trailing zeroes in the hashed value is less than the mask.
```

Comment:
I think "the hashed value is already in the sketch" is a bit misleading.
It's not necessary that this value is already inserted, it's just that the trailing zeros are less than the "level".

Reply:
Updated.